dom/src/makeDOMDriver.ts

```diff
@@ -1,75 +1,97 @@
 import {StreamAdapter} from '@cycle/base';
-import {init} from 'snabbdom';
+import {init as initSnabbdom, PatchFunction} from 'snabbdom';
 import xs, {Stream} from 'xstream';
 import {DOMSource} from './DOMSource';
 import {MainDOMSource} from './MainDOMSource';
 import {VNode} from './interfaces';
 import {VNodeWrapper} from './VNodeWrapper';
 import {getElement} from './utils';
 import defaultModules from './modules';
-import {IsolateModule} from './isolateModule';
+import {IsolateModule} from './IsolateModule';
 import {makeTransposeVNode} from './transposition';
 import {EventDelegator} from './EventDelegator';
-import xsSA from '@cycle/xstream-adapter';
+import xsAdapter from '@cycle/xstream-adapter';
 let MapPolyfill: typeof Map = require('es6-map');
 
-function makeDOMDriverInputGuard(modules: any) {
+export {makeDOMDriver}
+
+export interface DOMDriverOptions {
+  modules?: Array<any>;
+  transposition?: boolean;
+}
+
+function makeDOMDriver(
+  container: string | Element,
+  options: {transposition?: boolean, modules?: Array<any>} = {}
+): Function {
+  const {transposition = false, modules = defaultModules} = options;
+
+  makeDOMDriverModulesGuard(modules);
+
+  const isolateModule = new IsolateModule(new MapPolyfill<string, Element>());
+  const eventDelegators: Map<string, EventDelegator> =
+    new MapPolyfill<string, EventDelegator>();
+
+  const patch: PatchFunction = initSnabbdom([isolateModule.createModule()]
+                                 .concat(modules));
+
+  const rootElement: Element = getElement(container);
+  const vnodeWrapper = new VNodeWrapper(rootElement);
+
+  function _DOMDriver(vNode$: Stream<VNode>,
```

Comment:
Two naming issues: `_DOMDriver` should be `DOMDriver` because this function may appear in the stack trace, and for debugging purposes it's a bit nicer if users see `DOMDriver`.

I would name `vNode$` just `vnode$` because we do that in many places. Check 2 lines above, we have `vnodeWrapper`.